Source/Plugins/Core/com.equella.core/js/tsrc/modules/SearchFacetsModule.ts

```
@@ -94,7 +98,10 @@ export const listClassifications = async (
 ): Promise<Classification[]> =>
   Promise.all(
     (await getFacetsFromServer()).map<Promise<Classification>>(
-      async (settings) => ({
+      async (settings, index) => ({
+        // We know IDs won't be undefined here, but due to its type being number | undefined,
+        // we have to do a nullish coalescing
+        id: settings.id ?? index,
```

Hmmm, so it's only undefined if it's dirty. We should not have any 'dirty' entries here, so perhaps we should throw an error. If we leave like this it will probably be fine, but if we ever get corrupt data this could compound the situation.

Up to you.

---

I prefer leave it like this for now.